### PR TITLE
{bio}[foss/2021b] purge_dups v1.2.5 w/ Python 3.9.6

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by branfosj** [comment]
The link errors are all SSE2 / SSE4.1 / AVX, but not AVX2. The `sed -i "s/-O3/$CFLAGS/g" makefile` means that `-march=native` is in `CFLAGS`, so this chunk of the makefile:
```
(KMC_MAIN_DIR)/raduls_sse2.o: $(KMC_MAIN_DIR)/raduls_sse2.cpp
    $(CC) $(CFLAGS) -msse2 -c $< -o $@
$(KMC_MAIN_DIR)/raduls_sse41.o: $(KMC_MAIN_DIR)/raduls_sse41.cpp
    $(CC) $(CFLAGS) -msse4.1 -c $< -o $@
$(KMC_MAIN_DIR)/raduls_avx.o: $(KMC_MAIN_DIR)/raduls_avx.cpp
    $(CC) $(CFLAGS) -mavx -c $< -o $@
$(KMC_MAIN_DIR)/raduls_avx2.o: $(KMC_MAIN_DIR)/raduls_avx2.cpp
    $(CC) $(CFLAGS) -mavx2 -c $< -o $@
```
gets compile lines with `-march=native -msse2`.

Testing a local (ubuntu laptop) build: if I take out the `-march=native` then it builds, whereas it fails with that in. The `-march=native` only needs removing from the `CFLAGS` line - it can remain in the `KMC_TOOLS_CFLAGS` and it still builds.

**Comment by boegel** [comment]
@branfosj Thanks for the lead!

Seems like it's got to do with this bit of code (from `kmer_counter/raduls_impl.h`):

```c
#if defined(__AVX2__)
#define RADULS_RADIX_SORT_FUNNAME RadixSortMSD_AVX2
#elif defined (__AVX__)
#define RADULS_RADIX_SORT_FUNNAME RadixSortMSD_AVX
#elif defined(__SSE4_1__)
#define RADULS_RADIX_SORT_FUNNAME RadixSortMSD_SSE41
#elif defined(__SSE2__)
#define RADULS_RADIX_SORT_FUNNAME RadixSortMSD_SSE2
#endif
```

Building with `-march=native` results in always setting `__AVX2__`, which explains the missing symbols...

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2725.swalot.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/db07e10a827129559e9c1156d5f8956c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3503.doduo.os - Linux RHEL 8.4, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/5d816e18fe95b5aae7f611f8df78bbca for a full test report.

**Comment by boegel** [comment]
@SebastianAchilles Tests are failing on `jsc-zen2` with:

```
.../software/binutils/2.37-GCCcore-11.2.0/bin/ld.gold: error: cannot find -lc
.../software/binutils/2.37-GCCcore-11.2.0/bin/ld.gold: error: cannot find -lpthread
```

Can you also install `glibc-static` on `jsc-zen2` nodes, which should fix this issue?

File: easybuild/easyconfigs/p/purge_dups/purge_dups-1.2.5-foss-2021b.eb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+]
+
+use_pip = True
+
+default_easyblock = 'MakeCp'
+
+default_component_specs = {
+    'sources': [SOURCE_TAR_GZ],
+}
+
```
Comment (branfosj) [inline easybuild/easyconfigs/p/purge_dups/purge_dups-1.2.5-foss-2021b.eb:25]:
```suggestion
default_component_specs = {
    'sources': [SOURCE_TAR_GZ],
}

As both components have `sources` defined this is not needed.
```

**Review by branfosj — CHANGES_REQUESTED** [review]

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 7 out of 7 (1 easyconfigs in total)
bear-pg0105u36b.bear.cluster - Linux RHEL 8.5, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/7f5129ea2548feafed5d201926a2876f for a full test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

**Comment by SebastianAchilles** [comment]
> @SebastianAchilles Tests are failing on `jsc-zen2` with:
> 
> ```
> .../software/binutils/2.37-GCCcore-11.2.0/bin/ld.gold: error: cannot find -lc
> .../software/binutils/2.37-GCCcore-11.2.0/bin/ld.gold: error: cannot find -lpthread
> ```
> 
> Can you also install `glibc-static` on `jsc-zen2` nodes, which should fix this issue?

`glibc-static` is now installed on `jsc-zen2`

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
node3114.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/cddaa8b43abc2f8fc6aa2a9c43bce6ef for a full test report.

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.5, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/9655cc90c89709c1390c1f8b0050a166 for a full test report.

File: easybuild/easyconfigs/p/purge_dups/purge_dups-1.2.5-foss-2021b.eb (hunk 1 above)
Comment (branfosj) [reply easybuild/easyconfigs/p/purge_dups/purge_dups-1.2.5-foss-2021b.eb]: f6ab536

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.5, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/616a2bea991ba19d0eac3ba5653e6cc3 for a full test report.

**Comment by branfosj** [comment]
Going in, thanks @boegel!

**Pull request merged**